Feature/base64 checks

hroest opened the pull request:
- a few minor style improvements and addition of comments in the Base64 part
- add important check that input size needs to be 3 or more chars, otherwise padding code will fail

cbielow commented on src/openms/include/OpenMS/FORMAT/Base64.h:
```diff
@@ -451,13 +451,15 @@ namespace OpenMS
     if (in == "")
```
this can be removed, since its redundant to
size < 3
below?!

cbielow commented on src/openms/include/OpenMS/FORMAT/Base64.h:
```diff
@@ -777,13 +779,15 @@ namespace OpenMS
     if (in == "")
```
same here

cbielow commented on src/openms/include/OpenMS/FORMAT/Base64.h:
```diff
@@ -451,13 +451,15 @@ namespace OpenMS
     if (in == "")
       return;
 
+    // The length of a base64 string is a always a multiple of 3
```
its rather a multiple of 4 (judging by the loop below)?

hroest commented:
@cbielow I have now corrected the comments and also added some more documentation about how the decoding really works. I hope that is now documented properly for future generations :-) I tried the direct mapping with the lookup table as this may potentially be faster (no funny adding subtracting business) but I could not really find a measurable difference. Maybe this is just not a bottleneck here and since we wont optimize too early, lets keep the printable characters in.

timosachsenberg commented:
+1
